### PR TITLE
Python 3 cutover

### LINKED ISSUE #305: Python3 support
Is there any effort toward porting this project to Python 3?

It doesn't look like `vivisect` is being ported to Python 3 which doesn't help.  I was just curious if it was on your radar.

### PR DESCRIPTION
**Pull request by mr-tz**
here we go...

closes #305

### CODE REVIEW

**Review by williballenthin — APPROVED** [review]
i think some changes are superfluous, but looks good!

File: floss/api_hooks.py [hunk 1]
```diff
@@ -274,16 +274,16 @@ def readStringAtRva(emu, rva, maxsize=None):
...
+    ret = b""
     while True:
         if maxsize and maxsize <= len(ret):
             break
         x = emu.readMemory(rva, 1)
-        if x == "\x00" or x is None:
+        if x == b"\x00" or x is None:
             break
-        ret.append(x)
+        ret += x
```
[the remark this answers is not in the document]
Comment (mr-tz) [reply floss/api_hooks.py:284]: borrowed from vivisect/PE/__init__.py
will improve here

File: floss/const.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 KILOBYTE = 1024
 MEGABYTE = 1024 * KILOBYTE
 MAX_FILE_SIZE = 16 * MEGABYTE
-SUPPORTED_FILE_MAGIC = {"MZ"}
+SUPPORTED_FILE_MAGIC = b"MZ"
```
[the remark this answers is not in the document]
Comment (mr-tz) [reply floss/const.py:4]: do you mean dict vs bytes?

**Comment by mr-tz** [comment]
removed almost all `list` calls (added by `futurize`?!)

File: floss/const.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (williballenthin) [reply floss/const.py:4]: `set` vs a single instance of `bytes`

File: floss/api_hooks.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (williballenthin) [reply floss/api_hooks.py:284]: its probably not a huge deal, unless the strings are large. but stylistically, its not good.

File: floss/const.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mr-tz) [reply floss/const.py:4]: ah, gotcha, `black` complained here

[the remark this answers is not in the document]
Comment (mr-tz) [reply floss/const.py]: addressed in ef94797

File: floss/api_hooks.py [hunk 3]
```diff
@@ -274,16 +274,16 @@ def readStringAtRva(emu, rva, maxsize=None):
...
         rva += 1
-    return "".join(ret)
+    return bytes(ret)
```
Comment (williballenthin) [inline floss/api_hooks.py:286]: 👍🏼

**Review by williballenthin — APPROVED** [review]

**Pull request merged**